File: spring-boot-tutorial-basics-configuration/src/main/java/com/in28minutes/springboot/tutorial/basics/application/configuration/WelcomeResource.java
package com.in28minutes.springboot.tutorial.basics.application.configuration;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.HashMap;
import java.util.Map;

@RestController
public class WelcomeResource {

    @Value("${welcome.message}")
    private String welcomeMessage;

    private final BasicConfiguration configuration;

    public WelcomeResource(BasicConfiguration configuration) {
        this.configuration = configuration;
    }

    @GetMapping("/welcome")
    public String retrieveWelcomeMessage() {
        // Complex Method
        return welcomeMessage;
    }

    @RequestMapping("/dynamic-configuration")
    public Map<String, Object> dynamicConfiguration() {
        // Not the best practice to use a map to store different types!
        var configMap = new HashMap<String, Object>();
        configMap.put("message", configuration.getMessage());
        configMap.put("number", configuration.getNumber());
        configMap.put("key", configuration.isValue());

        return configMap;
    }
}

File: spring-boot-kotlin-basics-configuration/src/main/kotlin/com/in28minutes/springboot/tutorial/basics/application/configuration/WelcomeResource.kt
package com.in28minutes.springboot.tutorial.basics.application.configuration

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.beans.factory.annotation.Value
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
class WelcomeResource {
    @Value("\${welcome.message}")
    private val welcomeMessage: String? = null

    @Autowired
    private val configuration: BasicConfiguration? = null

    @GetMapping("/welcome")
    fun retrieveWelcomeMessage(): String? {
        // Complex Method
        return welcomeMessage
    }

    @RequestMapping("/dynamic-configuration")
    fun dynamicConfiguration(): Map<String, Any?> {
        // Not the best practice to use a map to store different types!
        val configMap: MutableMap<String, Any?> = HashMap()
        configMap["message"] = configuration!!.message
        configMap["number"] = configuration.number
        configMap["key"] = configuration.isValue
        return configMap
    }
}